### PR TITLE
feat: add support for deadman mode

### PR DESCRIPTION
**Pull request by carlosingles**
This PR also changes the endpoint that is used for retrieving the high score values. Instead of using the CSV version, it now uses the json version which means upstream changes to the osrs high scores API will no longer impact the response parsing of this package.

I have made sure that it is not a breaking change, since the names of the values returned in the json response differ to what is defined in this package.

### CODE REVIEW

File: src/hiscores/hiscores.module.ts [hunk 1]
```diff
@@ -12,47 +12,30 @@ import { Bosses, Minigames, Player, Skills } from './player.model';
...
+    const bosses = {} as Bosses;
+    for (const activity of data.activities) {
+        const { rank, score, name } = activity;
+        const isMinigame = !!MINIGAME_NAMES[name];
+        const activityName = isMinigame ? MINIGAME_NAMES[name] : BOSS_NAMES[name];
+        if (activityName) {
+            if (isMinigame) minigames[activityName] = { rank, score };
+            else bosses[activityName] = { rank, kills: score };
+        }
+    }
```
Comment (atjeff) [inline src/hiscores/hiscores.module.ts]:
```suggestion
    for (const activity of data.activities) {
        const { rank, score, name } = activity;

        if (!!MINIGAME_NAMES[name]) {
            minigames[MINIGAME_NAMES[name]] = { rank, score };
        } else if (!!BOSS_NAMES[name]) {
            bosses[BOSS_NAMES[name]] = { rank, kills: score };
        }
    }
```

How do you feel about this change?

**Comment by atjeff** [comment]
@carlosingles Thanks so much for your continued interest in this work I haven't touched in so long, it means a lot that you're keeping it up to date! 

I'm actually really shocked to see Jagex release a json endpoint, is this new? 

The changes look good - I made one suggestion. I also see an out of date test in the spec, you can update this with `yarn test -u` or `npm run test -u`

**Comment by carlosingles** [comment]
No worries at all 🙂 

I use your package to scrape the high scores API on a daily schedule, which is why I usually notice that an update is required whenever Jagex updates their dataset.

The way I found out this endpoint even existed was due to this API change not being documented on the [wiki page](https://runescape.wiki/w/Application_programming_interface#Hiscores_Lite_2) that I was using to determine what had changed. 

I noticed that runelite was still reporting stats correctly and that's when I found the endpoint in their [source](https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/hiscore/HiscoreEndpoint.java#L36) (looks to be added 6 months ago)

**Review by atjeff — APPROVED** [review]

**Comment by atjeff** [comment]
Should be released as 2.2.5

**Pull request merged**